comixed-web/src/app/services/session.service.ts

```diff
@@ -23,12 +23,33 @@ import { SessionUpdateRequest } from '@app/models/net/session-update-request';
 import { LOAD_SESSION_UPDATE_URL } from '@app/app.constants';
 import { HttpClient } from '@angular/common/http';
 import { LoggerService } from '@angular-ru/logger';
+import { TaskCountService } from '@app/services/state/task-count.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class SessionService {
-  constructor(private logger: LoggerService, private http: HttpClient) {}
+  constructor(
+    private logger: LoggerService,
+    private http: HttpClient,
+    private taskCountService: TaskCountService
+  ) {}
+
+  /**
+   * Invoked to start subscriptions.
+   */
+  startSubscriptions(): void {
+    this.logger.debug('Starting session services');
+    this.taskCountService.start();
```

Comment:
Why does sessionService.startSubscriptions call taskCountService.start? Is this some kind of test to see if it works well? If not, what is the purpose of this task counter normally used for import count service? ?

Reply:
The goal there is to divide up all of the various parts of the session that will use the messaging bus. Rather than having SessionService have some huge, monolithic method that subscribes to task counts, comic changes, reading list changes, etc., we'll code up each type of subscription as a separate service type (like TaskCountService) that can be started and stopped by SessionService.

That will also make it easier to test them since, like TaskCountService, they are isolated and testable by themselves.